frontend/apps/web/app/(mgmt)/[account]/new/job/connect/page.tsx

```diff
@@ -576,3 +718,76 @@ export default function Page({ searchParams }: PageProps): ReactElement {
     </div>
   );
 }
+
+interface TestConnectionBadgeProps {
+  isValidating: boolean;
+  validationResponse: CheckConnectionConfigResponse | undefined;
+  id: string | undefined;
+  accountName: string;
+}
+
+function TestConnectionBadge(props: TestConnectionBadgeProps): ReactElement {
+  const { isValidating, validationResponse, id, accountName } = props;
+
+  const url = `/${accountName}/connections/${id}/permissions`;
+
+  const handleValidationResponse = (
+    vr: CheckConnectionConfigResponse | undefined
+  ): ReactElement => {
+    if (vr && vr?.isConnected && vr.privileges.length > 0) {
+      return (
+        <CheckCircledIcon className="h-4 w-4 text-green-600 dark:text-green-400" />
+      );
+    } else if (vr && vr?.isConnected && vr.privileges.length === 0) {
+      return (
+        <div className="relative flex items-center">
+          <Badge variant="outline" className="flex items-center absolute">
+            <div className="flex flex-row items-center gap-2">
+              <IoWarning className="h-4 w-4 text-yellow-600 dark:text-orange-400" />
+              <div className="text-nowrap text-xs">
+                Warning - No tables found.{' '}
+                <a href={url} className="underline" target="_blank">
+                  More info
+                </a>
+              </div>
+              <ArrowTopRightIcon className="h-4 w-4" />
+            </div>
+          </Badge>
+        </div>
+      );
+    } else if (vr && !vr.isConnected) {
+      return (
+        <div className="relative flex items-center">
+          <Badge variant="outline" className="flex items-center absolute">
+            <div className="flex flex-row items-center gap-2">
+              <ExclamationTriangleIcon className="h-4 w-4 text-red-500" />
+              <div className="text-nowrap text-xs">
+                Error - Unable to connect.{' '}
+                <a href={url} className="underline" target="_blank">
+                  More info
+                </a>
+              </div>
+              <ArrowTopRightIcon className="h-4 w-4" />
+            </div>
+          </Badge>
+        </div>
+      );
+    } else {
+      return <div></div>;
+    }
+  };
+
+  return (
+    <div className="relative flex items-center">
+      <div className="absolute">
+        {id && isValidating ? (
+          <Spinner className="text-black dark:text-white" />
+        ) : id && !isValidating ? (
+          <div>{handleValidationResponse(validationResponse)}</div>
```

Comment:
let's move that internal component to be its own module-level function.
```suggestion
          <div><TestConnectionValidationResponse ...props... />
```